activerecord/lib/active_record/reflection.rb

```diff
@@ -302,6 +307,17 @@ def has_cached_counter?
           active_record.has_attribute?(counter_cache_column)
       end
 
+      # Returns whether this association has a counter cache and its column values were backfilled
+      # (and so it is used internally by methods like +size+/+any?+/etc).
+      def has_active_cached_counter?
+        return false unless has_cached_counter?
+
+        counter_cache = options[:counter_cache] ||
+                        (inverse_which_updates_counter_cache && inverse_which_updates_counter_cache.options[:counter_cache])
+
+        !counter_cache.is_a?(Hash) || counter_cache[:active] != false
```

Comment:
I'd prefer if we normalized `options[:counter_cache]` in the initializer.

Reply:
To clarify what I mean, if you pass `counter_cache: true` or `counter_cache: :column_name`, we sould convert it to the equivalent `Hash` so that then we get rid of these `is_a?` and always deal with a Hash (or even a small PORO, might as well).

Reply:
Updated with the suggestion. Please take a look.